### DISCUSSION TITLE
Z-Push with Mailcow

### DISCUSSION THREAD

**Discussion by gmcpaul** [Z-Hub/Z-Push#38] [Q&A] [answered]
Hi, 
i would like to test Z-Push with mailcow system.
i lastly used it with Kopano which was working perfectly.
Can you maybe give me info or tipps how to configure these settings?
im trying to use it via imap, caldav, carddav

so far i think i found the proper settings for caldav and carddav but for imap im not aware how to deal with the authentication.

has anyone fiddled around with that constellation so far?

ty,
paul

**Comment by gmcpaul** [comment]
https://docs.mailcow.email/client/client-manual/

CalDAV https://mail.example.com/SOGo/dav/user@example.com/Calendar/personal/
CardDAV https://mail.example.com/SOGo/dav/user@example.com/Contacts/personal/

should be the settings for the caldav and carddav which point to sogo , but imap is somehow of a riddle for me..

**Reply by matidau** [reply, marked as answer]
Hi @gmcpaul,

For IMAP this link has a basic setup: https://kittmedia.com/en/2021/debian-setup-z-push-with-imap-and-nginx/ 
If you are looking for something more indepth I would look at the Mailinabox configuration: https://github.com/mail-in-a-box/mailinabox/blob/main/conf/zpush/backend_imap.php

Cheers,
Mat

**Comment by gmcpaul** [comment]
@matidau 
thanks i got the proper values and communication and first sync was working.
now after resyncing i got another issue 

it seems that the syncstates do not get written or something like that


==> /var/log/z-push/z-push.log <==
21/12/2023 15:13:55 [3466059] [ INFO] [bla@bla.org] cmd='Sync' memory='1.96 MiB/4.00 MiB' time='0.46s' devType='WindowsOutlook' devId='291a3e9a66fe4ea4a2b7e904b1442fe4' getUser='bla@bla.org' from='xxx.xxx.xxx.xxx' idle='0s' version='2.7.1
' method='POST' httpcode='200'
21/12/2023 15:13:56 [3488689] [ INFO] [bla@bla.org] NoHierarchyCacheAvailableException: Folderid '16cd8dcf' is not fully synchronized on the device - code: 0 - file: /usr/share/z-push/lib/core/devicemanager.php:466
21/12/2023 15:13:56 [3488689] [ INFO] [bla@bla.org] HandlePing(): unknown collection '16cd8dcf', triggering HierarchySync


==> /var/log/z-push/z-push-error.log <==
21/12/2023 15:13:55 [3466059] [WARN] [bla@bla.org] Not possible to determine class of request. Request did not contain class and apparently there is an issue with the HierarchyCache.


is there a known problem to this or how to solve it?

thanks,
paul

**Reply by matidau** [reply]
Hi Paul,

~~There is a similar report on issue #40~~

~~Can you confirm that you are using Outlook as the client and the exact version it is?~~

There is a similar error message on the Kopano Community Forums, can you review and let me know if this matches your experience? 

https://forum.kopano.io/topic/3924/outlook-mail-folders-are-empty-nohierarchycacheavailableexception/8

Cheers,
Mat

**Reply by gmcpaul** [reply]
hi mat im sorry but i changed now to groupware office, which is using the zpush env as well but it works there somehow...

**Reply by matidau** [reply]
Great to hear 😊
